pkg/skaffold/deploy/kubectl/cli.go

```diff
@@ -73,6 +75,50 @@ func (c *CLI) Apply(ctx context.Context, out io.Writer, manifests ManifestList)
 	return nil
 }
 
+type getResult struct {
+	Items []resource `json:"items"`
+}
+
+type resource struct {
+	Metadata metadata `json:"metadata"`
+}
+
+type metadata struct {
+	Name              string `json:"name"`
+	DeletionTimestamp string `json:"deletionTimestamp"`
+}
+
+// GetPendingDeletions uses `kubectl get` to list resources that are marked for deletion.
+func (c *CLI) GetPendingDeletions(ctx context.Context, manifests ManifestList) ([]string, error) {
+	args := []string{"-f", "-", "--ignore-not-found=true", "-ojson"}
+
+	var out bytes.Buffer
+	if err := c.Run(ctx, manifests.Reader(), &out, "get", c.args(c.Flags.Global, args...)...); err != nil {
+		return nil, fmt.Errorf("kubectl get: %w", err)
+	}
+
+	if out.Len() == 0 {
+		// No resource found.
+		return nil, nil
+	}
+
+	var result getResult
+	if err := json.Unmarshal(out.Bytes(), &result); err != nil {
+		return nil, err
+	}
+
+	var list []string
+
+	// Find resources with `metadata.deletionTimestamp` field.
+	for _, r := range result.Items {
+		if r.Metadata.DeletionTimestamp != "" {
+			list = append(list, r.Metadata.Name)
```

Comment:
I think it's worth appending the group/kind too.  I think there's some code in the `pkg/diag` that can be used here.